### PR TITLE
Fix refusal search by case

### PR DESCRIPTION
**Pull request by AdeHarrison**
# Motivation and Context
case should be located using case id

# What has changed
Case located by case id instead of questionnaire id

# How to test?
checkout branch and build local docker image along with any appropriate attached PRs.
run using docker-dev
run tests

# Links
[Trello](https://trello.com/c/qCIrbvPi/1018-case-processor-refusal-event-structure-is-incorrect)

# Screenshots (if appropriate):

### CODE REVIEW

**Comment by AdeHarrison** [comment]
[Case Processor - Refusal event case lookup is incorrect (3?)](https://trello.com/c/qCIrbvPi/1018-case-processor-refusal-event-structure-is-incorrect)

File: src/main/java/uk/gov/ons/census/casesvc/service/RefusalProcessor.java [hunk 1]
```diff
@@ -3,62 +3,49 @@
...
     caseProcessor.emitCaseUpdatedEvent(caze);
 
     eventLogger.logRefusalEvent(
-        uacQidLink,
-        REFUSAL_RECEIVED,
-        EventType.CASE_UPDATED,
-        refusal,
-        refusalEvent.getEvent(),
-        refusal.getResponseDateTime());
+        caze, REFUSAL_RECEIVED, EventType.CASE_UPDATED, refusal, refusalEvent.getEvent());
```
Comment (davidmort) [inline src/main/java/uk/gov/ons/census/casesvc/service/RefusalProcessor.java]:
```suggestion
        caze, REFUSAL_RECEIVED, EventType.REFUSAL_RECEIVED, refusal, refusalEvent.getEvent());
```

Comment (AdeHarrison) [reply src/main/java/uk/gov/ons/census/casesvc/service/RefusalProcessor.java]: done

File: src/main/java/uk/gov/ons/census/casesvc/logging/EventLogger.java [hunk 2]
```diff
@@ -55,20 +56,23 @@ public void logReceiptEvent(
...
+    loggedEvent.setId(UUID.randomUUID());
+    loggedEvent.setCaze(caze);
+    loggedEvent.setCaseId(UUID.fromString(payload.getCollectionCase().getId()));
+    loggedEvent.setEventDate(event.getDateTime());
+    loggedEvent.setRmEventProcessed(OffsetDateTime.now());
+    loggedEvent.setEventDescription(eventDescription);
+    loggedEvent.setEventType(eventType);
+    loggedEvent.setEventChannel(event.getChannel());
+    loggedEvent.setEventSource(event.getSource());
+    loggedEvent.setEventTransactionId(UUID.randomUUID());
```
Comment (davidmort) [inline src/main/java/uk/gov/ons/census/casesvc/logging/EventLogger.java]: this should be the transaction id that comes in on the message

Comment (liamtoozer) [reply src/main/java/uk/gov/ons/census/casesvc/logging/EventLogger.java]: Think this should be resolved from [this bit](https://github.com/ONSdigital/census-rm-case-processor/blob/master/src/main/java/uk/gov/ons/census/casesvc/logging/EventLogger.java#L112-L116) after merge conflicts are resolved in this PR

**Comment by davidmort** [comment]
LGTM after merge conflicts have been sorted 👍

File: src/test/java/uk/gov/ons/census/casesvc/logging/EventLoggerTest.java [hunk 3]
```diff
@@ -159,27 +161,23 @@ public void testLogReceiptEvent() {
...
 
     // When
     underTest.logRefusalEvent(
-        new UacQidLink(),
-        "TEST_LOGGED_EVENT",
-        EventType.UAC_UPDATED,
-        expectedRefusal,
-        new EventDTO(),
-        OffsetDateTime.now());
+        new Case(), "TEST_LOGGED_EVENT", EventType.UAC_UPDATED, expectedRefusal, new EventDTO());
```
Comment (ryangrundy7) [inline src/test/java/uk/gov/ons/census/casesvc/logging/EventLoggerTest.java]: Shouldn't this be `EventType.REFUSAL_RECEIVED?`

Comment (AdeHarrison) [reply src/test/java/uk/gov/ons/census/casesvc/logging/EventLoggerTest.java]: done

File: src/main/java/uk/gov/ons/census/casesvc/logging/EventLogger.java (hunk 2 above)
Comment (davidmort) [reply src/main/java/uk/gov/ons/census/casesvc/logging/EventLogger.java]: doesn't look this has been fixed by the conflict merges???

**Pull request merged**